system_metrics_collector/test/system_metrics_collector/test_functions.hpp

```diff
@@ -0,0 +1,196 @@
+#include <mutex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "metrics_statistics_msgs/msg/metrics_message.hpp"
+#include "metrics_statistics_msgs/msg/statistic_data_type.hpp"
+
+#include "moving_average_statistics/moving_average.hpp"
+
+#include "system_metrics_collector/constants.hpp"
+#include "system_metrics_collector/linux_memory_measurement_node.hpp"
+#include "system_metrics_collector/utilities.hpp"
+
+
+using metrics_statistics_msgs::msg::MetricsMessage;
+using metrics_statistics_msgs::msg::StatisticDataPoint;
+using metrics_statistics_msgs::msg::StatisticDataType;
+
+namespace test_functions
+{
+
+using ExpectedStatistics =
+  std::unordered_map<decltype(StatisticDataPoint::data_type), decltype(StatisticDataPoint::data)>;
+
+/**
+ * Convert input StatisticData to a map of StatisticDataPoint::data_type ->
+ * StatisticDataPoint::data.
+ *
+ * @param src input data to convert
+ * @return output ExpectedStatistics to use for testing.
+ */
+ExpectedStatistics StatisticDataToExpectedStatistics(
+  const moving_average_statistics::StatisticData & src)
+{
+  ExpectedStatistics expected{};
+  expected[StatisticDataType::STATISTICS_DATA_TYPE_AVERAGE] = src.average;
+  expected[StatisticDataType::STATISTICS_DATA_TYPE_MINIMUM] = src.min;
+  expected[StatisticDataType::STATISTICS_DATA_TYPE_MAXIMUM] = src.max;
+  expected[StatisticDataType::STATISTICS_DATA_TYPE_STDDEV] = src.standard_deviation;
+  expected[StatisticDataType::STATISTICS_DATA_TYPE_SAMPLE_COUNT] = src.sample_count;
+  return expected;
+}
+
+/**
+ * Check statistic equality. Fails for any unknown statistic type
+ *
+ * @param expected_stats expected data
+ * @param actual data from a received ROS2 message
+ */
+void ExpectedStatisticEquals(
+  const ExpectedStatistics & expected_stats,
+  const metrics_statistics_msgs::msg::MetricsMessage & actual)
+{
+  for (const StatisticDataPoint & stats_point : actual.statistics) {
+    const auto type = stats_point.data_type;
+    switch (type) {
+      case StatisticDataType::STATISTICS_DATA_TYPE_SAMPLE_COUNT:
+        EXPECT_DOUBLE_EQ(expected_stats.at(type), stats_point.data) << "unexpected sample count";
+        break;
+      case StatisticDataType::STATISTICS_DATA_TYPE_AVERAGE:
+        EXPECT_DOUBLE_EQ(expected_stats.at(type), stats_point.data) << "unexpected average";
+        break;
+      case StatisticDataType::STATISTICS_DATA_TYPE_MINIMUM:
+        EXPECT_DOUBLE_EQ(
+          expected_stats.at(type), stats_point.data) << "unexpected min";
+        break;
+      case StatisticDataType::STATISTICS_DATA_TYPE_MAXIMUM:
+        EXPECT_DOUBLE_EQ(
+          expected_stats.at(type), stats_point.data) << "unexpected max";
+        break;
+      case StatisticDataType::STATISTICS_DATA_TYPE_STDDEV:
+        EXPECT_DOUBLE_EQ(expected_stats.at(type), stats_point.data) << "unexpected stddev";
+        break;
+      default:
+        FAIL() << "received unknown statistics type: " << std::to_string(type);
```

Comment:
```suggestion
        FAIL() << "received unknown statistics type: " << std::dec << static_cast<unsigned int>(type);
```
The `StatisticDataType` constants are all `uint8`, so `to_string` will treat it like a character. Since this is a stream, you can use `std::dec` to switch to base10 formatting